Objects/longobject.c

```diff
@@ -5260,6 +5260,35 @@ long_is_finite(PyObject *v)
 }
 #endif
 
+/*[clinic input]
+int.as_integer_ratio
+
+Return integer ratio.
+
+Return a pair of integers, whose ratio is exactly equal to the original int
+and with a positive denominator.
+
+>>> (10).as_integer_ratio()
+(10, 1)
+>>> (-10).as_integer_ratio()
+(-10, 1)
+>>> (0).as_integer_ratio()
+(0, 1)
+[clinic start generated code]*/
+
+static PyObject *
+int_as_integer_ratio_impl(PyObject *self)
+/*[clinic end generated code: output=e60803ae1cc8621a input=55ce3058e15de393]*/
+{
+  if PyLong_CheckExact(self)
+    return PyTuple_Pack(2, self, _PyLong_One);
+  else {
+      PyObject *temp = PyNumber_Positive(self);
+      Py_DECREF(temp);
```

Comment:
The DECREF needs to occur *after* building the tuple; otherwise, the Python integer object can (and likely will) disappear before it gets used.